code/game/machinery/aug_manipulator.dm

```diff
@@ -0,0 +1,131 @@
+/obj/machinery/aug_manipulator
+	name = "\improper augment manipulator"
+	desc = "A machine for custom fitting augmentations, with in-built spraypainter."
+	icon = 'icons/obj/pda.dmi'
+	icon_state = "pdapainter"
+	density = TRUE
+	anchored = TRUE
+	obj_integrity = 200
+	max_integrity = 200
+	var/obj/item/bodypart/storedpart
+	var/initial_icon_state
+	var/static/list/style_list_icons = list("standard" = 'icons/mob/augmentation/augments.dmi', "engineer" = 'icons/mob/augmentation/augments_engineer.dmi', "security" = 'icons/mob/augmentation/augments_security.dmi', "mining" = 'icons/mob/augmentation/augments_mining.dmi')
+
+
+/obj/machinery/aug_manipulator/Initialize()
+    initial_icon_state = initial(icon_state)
+    return ..()
+
+/obj/machinery/aug_manipulator/update_icon()
+	cut_overlays()
+
+	if(stat & BROKEN)
+		icon_state = "[initial_icon_state]-broken"
+		return
+
+	if(storedpart)
+		add_overlay("[initial_icon_state]-closed")
+
+	if(powered())
+		icon_state = initial_icon_state
+	else
+		icon_state = "[initial_icon_state]-off"
+
+/obj/machinery/aug_manipulator/Destroy()
+	QDEL_NULL(storedpart)
+	return ..()
+
+/obj/machinery/aug_manipulator/on_deconstruction()
+	if(storedpart)
+		storedpart.forceMove(loc)
+		storedpart = null
+
+/obj/machinery/aug_manipulator/contents_explosion(severity, target)
+	if(storedpart)
+		storedpart.ex_act(severity, target)
+
+/obj/machinery/aug_manipulator/handle_atom_del(atom/A)
+	if(A == storedpart)
+		storedpart = null
+		update_icon()
+
+/obj/machinery/aug_manipulator/attackby(obj/item/O, mob/user, params)
+	if(default_unfasten_wrench(user, O))
+		power_change()
+		return
+
+	else if(istype(O, /obj/item/bodypart))
```

Comment:
@RemieRichards Not sure how else to go about it since cyborg parts inherit individually from every single regular bodypart